Separate `Your account` settings to two sections - `Profile` and `Account & privacy`.

sahil839 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
This PR separtes the `Your account` settings to two sections - `Profile` and `Account & privacy`.

`Profile` section contains - user name, profile picture, custom profile fields
`Account & privacy` section contains - email, password, role, deactivate button and api key section.

Have also added some prep commits for fixing class names and live-update bugs.

https://chat.zulip.org/#narrow/stream/101-design/topic/privacy.2Fsecurity.20settings

Fixes #18848.

 <!-- How have you tested? -->


**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
![Screenshot from 2021-06-25 16-26-53](https://user-images.githubusercontent.com/35494118/123652817-41ea8480-d84a-11eb-9d4d-a1f09e88bfd0.png)

![Screenshot from 2021-06-25 16-26-58](https://user-images.githubusercontent.com/35494118/123652829-44e57500-d84a-11eb-894e-dec603c2c4dc.png)


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

sahil839 commented:
We still need to discuss what to do when there are no custom profile fields as the `Profile` page will only contain `name` field in that case. Also we need to discuss about the location of `Preview profile` button as Alya suggested to place it below `Profile picture` in case of large number of custom profile fields. 

Can have these discussion on czo but the prep commits can still be merged.

sahil839 commented:
Docs changes are still need to be done, but we will merge them after having a final decision on design.

sahil839 commented on templates/zerver/help/edit-your-profile.md:
```diff
@@ -12,16 +12,12 @@ fill them out.
-!!! warn ""
-    Note: The **Profile** section will only appear if an organization
-    administrator has added custom profile fields.
-
```
I am not sure what to write here as we have a single `Profile` section which contains name field also, so just removed it for now.

timabbott replied:
Yeah, I think deleting this is correct.

timabbott commented:
> We still need to discuss what to do when there are no custom profile fields as the Profile page will only contain name field in that case. Also we need to discuss about the location of Preview profile button as Alya suggested to place it below Profile picture in case of large number of custom profile fields.

For this, I think the right plan is to migrate a couple fields that appear on the profile to this page:
* Timezone (from display settings), which is actually only used for the user's profile
* A read-only display of the user's "Joined on " date, similar to how Role appears on "Account and privacy".

With those changes, the minimum height of the left section will look more like this:

![image](https://user-images.githubusercontent.com/2746074/123823291-e6ec7800-d8b1-11eb-9355-c98dec9abe27.png)

Which looks a lot more reasonable.  Once we've merged that, we can also look at adding an advertisement for setting new custom profile fields and/or 1-2 fields that are created for new organizations by default as follow-ups.

timabbott commented:
I merged the commits through e0fc6df2b42cad196ff0dbca1ae4f98eafb1ee55.  Can you explain the avatar live update commit a bit more?  I don't understand that change yet.

sahil839 commented:
Re the avatar live update commit, we mostly (probably all except this) use add `disabled` prop on buttons only and we add the css for buttons (and some specific class elements) when disabled as `cursor: not-allowed` or `pointer-events: none` which means the user cannot use these buttons when disabled. This is not the case for the avatar widget as it we use a div element there and not button and nor those specific classes. Also, since we use the avatar in two ways - showing image and a button also and the div with class `image_upload_button` is used for click handler for uploading, so hiding (adding `display: none`) will mean that it will not be visible on hover and also cannot be clicked.

Does that make sense? I will update the commit message if so.

timabbott commented:
Yeah, that makes sense; a commit message update plus perhaps a small comment noting that we disable this widget by just hiding its edit UI makes sense.  

I think we also should have a `?` element with tooltip similar to what we have for "Full name" explaining that "Changing your avatar is disabled in this organization.", for consistency with similar features.

sahil839 commented:
@timabbott updated the commit message and added comments for live-update of avatar widget commit.
Also added tooltip to the right of `profile picture` heading which will be visible only if user is not allowed to change avatar.
![Screenshot from 2021-06-30 15-30-25](https://user-images.githubusercontent.com/35494118/123964663-4d1bec80-d9d1-11eb-80f1-33ce44ce8d7d.png)
![Screenshot from 2021-06-30 15-30-20](https://user-images.githubusercontent.com/35494118/123964673-4f7e4680-d9d1-11eb-865a-75351f067a28.png)

timabbott commented:
Cool, merged the last two prep commits, through efa19597bb0d4bce5ed25b9e3e8ba9c2cfa2af16.

Let me know when you've added the extra commit(s) to fill out the "Profile" page a bit more for organizations without custom profile fields.

timabbott commented:
This is great, merged after the following changes:
* Moved role to the "profile" page, since that gets displayed.
* Moved "Date joined" to be situated above the timezone, so that the timezone effectively works like a first custom profile field from a visual perspective.
* Renamed "User settings" to "Account" for the email/password/deactivate block.

I think we should migrate the "presence" notification setting to a "Privacy" section there ASAP, just because the page now feels pretty weird labeled "Account and privacy" with no privacy-related settings on it :)

Thanks @sahil839!
